### PR TITLE
Component label

### PR DESCRIPTION
**Pull request by shmbajaj**
## Label Editor Component
- Label Editor component is composed of three components LabelsEditorHeader, AddLabels, GetLablels and
uses them to show labels, add labels, delete labels and save labels for a note.
- Label Editor component is available locally for notes only, not at gloabl context for all available notes.
- LabelsEditorHeader component is used for closing labels editor view and saving labels.
- AddLabels is used to add label to label list for a note.
- GetLabels is used to render list of notes and gives power of deleting label.
- To add or edit labels click on label icon on action below note editor, which will open lable editor view and 
to close click on icon on top right corner.

## What files to check
- src/components/LabelsEditor/*
- src/components/Main/ActionBar/ActionBar.js

```
Actual Files to check: 4-5
```

## Preview

https://user-images.githubusercontent.com/29247011/161734566-ce8a4c0c-9502-4f77-8634-8527acfabc99.mp4

## Deploy Preview
https://deploy-preview-5--commonplace-book.netlify.app/

### CODE REVIEW

File: src/components/FeatureBar/Feature/Feature.styles.module.css [hunk 1]
```diff
@@ -3,15 +3,36 @@
   padding-left: var(--size-11);
   background: none;
   border: 0;
+  border-bottom: 1px solid;
 }
 
 .feature:hover {
-  background-color: var(--color-3a3a3a);
+  /* background-color: var(--color-3a3a3a); */
+  background-color: var(--color-e5e5e5);
```
Comment (JowelTisso) [inline src/components/FeatureBar/Feature/Feature.styles.module.css:11]: Remove unsued code

**Review by JowelTisso — APPROVED** [review]
** Too big PR
* Try keeping your PR small

File: src/components/FeatureBar/Feature/Feature.styles.module.css (hunk 1 above)
Comment (shmbajaj) [reply src/components/FeatureBar/Feature/Feature.styles.module.css:11]: > * Too big PR
> * Try keeping your PR small

This I will not repeat from next project.

Comment (shmbajaj) [reply src/components/FeatureBar/Feature/Feature.styles.module.css:11]: I will remove commented code.

File: src/components/LabelsEditor/LabelsEditor.components..js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  return (
+    <div className={styles["labelEditorInputArea"]}>
+      <p></p>
+      <input
+        type="text"
+        placeholder="Add a new label"
+        className={styles["labelInputArea"]}
+        value={label.labelText}
+        onChange={(e) => onChangeHandler(e)}
+      />
```
Comment (JowelTisso) [inline src/components/LabelsEditor/LabelsEditor.components..js:84]: You can just pass the function name here

File: src/components/Main/ActionBar/ActionBar.js [hunk 3]
```diff
@@ -1,41 +1,168 @@
...
+    }
+
+    if (featureName === "archives" && iconName === "archive") {
+      return "unarchive";
+    }
+
+    if (featureName === "archives" && iconName !== "archive") {
+      return iconName;
+    }
+  }
```
Comment (JowelTisso) [inline src/components/Main/ActionBar/ActionBar.js:117]: You can extract these strings as constants if it has multiple use

**Review by JowelTisso — APPROVED** [review]
* Good code structure
* Rest Looks Good

File: src/components/Main/ActionBar/ActionBar.js (hunk 3 above)
Comment (shmbajaj) [reply src/components/Main/ActionBar/ActionBar.js:117]: These are local to here but will be refactored as helpers.

File: src/components/LabelsEditor/LabelsEditor.components..js (hunk 2 above)
Comment (shmbajaj) [reply src/components/LabelsEditor/LabelsEditor.components..js:84]: Okay, this I don't know.

**Pull request merged**